kcp/controllers/watcher_controller.go

```diff
@@ -283,8 +255,8 @@ func (r *WatcherReconciler) createOrUpdateIstioVirtualServiceForCR(ctx context.C
 		return nil
 	}
 	// check if config already exists
-	if util.IsVirtualServiceConfigChanged(listenerVirtualService, obj, istioGatewayResourceName) {
-		util.UpdateVirtualServiceConfig(listenerVirtualService, obj, istioGatewayResourceName)
+	if util.IsVirtualServiceConfigChanged(listenerVirtualService, obj, IstioGatewayResourceName, IstioGatewayNamespace) {
+		util.UpdateVirtualServiceConfig(listenerVirtualService, obj, IstioGatewayResourceName, IstioGatewayNamespace)
```

Comment:
who is going to change this virtual service?

Even if, please use the kubebuilder watch / owns mechanis.

Reply:
The virtual service config may change if the watcher CR changes.
I agree. Using `watch` mechanism is a better solution 👍